Style and grammar review for the OpenID Connect authorization code flow (#53338)
quarkusio/quarkus: docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc

# Use RS512 signature algorithm instead of the default RS256
quarkus.oidc.credentials.jwt.signature-algorithm=RS512

# The token endpoint URL is the default audience value; use the base address URL instead:
quarkus.oidc.credentials.jwt.audience=${quarkus.oidc-client.auth-server-url}

> # custom subject instead of the client ID:

Reviewer: ```suggestion
# custom subject instead of the client id:
```

When the user is redirected back to Quarkus to complete the authorization code flow, Quarkus expects the request URI to include the `state` query parameter and that it matches the current state cookie value.

The default state cookie age is 5 minutes, and you can change it with a `quarkus.oidc.authentication.state-cookie-age` Duration property.

Quarkus creates a unique state cookie name every time a new authorization code flow is started to support multi-tab authentication. Many concurrent authentication requests on behalf of the same user may cause many state cookies to be created.

If you do not want to allow your users to use multiple browser tabs to authenticate, it is recommended to disable this with `quarkus.oidc.authentication.allow-multiple-code-flows=false`. It also ensures that the same state cookie name is created for every new user authentication.

[[token-state-manager]]
==== Session cookie and default TokenStateManager

OIDC `CodeAuthenticationMechanism` uses the default `io.quarkus.oidc.TokenStateManager` interface implementation to keep the ID, access, and refresh tokens returned in the authorization code or refresh grant responses in an encrypted session cookie.

> It makes Quarkus OIDC endpoints completely stateless, and it is recommended to follow this strategy to achieve the best scalability results.

Reviewer 2: ```suggestion
It makes Quarkus OIDC endpoints completely stateless, and the preferred approach is to follow this strategy to achieve the best scalability results.
```

# Auth0 does not return the `end_session_endpoint` metadata property. Instead, you must configure it:
quarkus.oidc.end-session-path=v2/logout
# Auth0 will not recognize the 'post_logout_redirect_uri' query parameter, so ensure it is named as 'returnTo':
quarkus.oidc.logout.post-logout-uri-param=returnTo

# Set more properties if needed.
# For example, if 'client_id' is provided, then a valid logout URI should be set as the Auth0 Application property, without it, as the Auth0 Tenant property:
quarkus.oidc.logout.extra-params.client_id=${quarkus.oidc.client-id}
----
====

[[back-channel-logout]]
==== Back-channel logout

The OIDC provider can force all applications to log out by using the authentication data.
This is known as back-channel logout.
In this case, OIDC will call a specific URL for each application to trigger the logout.

> OIDC providers use the link:https://openid.net/specs/openid-connect-backchannel-1_0.html[Back-channel logout] to log out the current user from all the applications they are currently logged into, bypassing the user agent.

Reviewer: ```suggestion
OIDC providers use the link:https://openid.net/specs/openid-connect-backchannel-1_0.html[Back-channel logout] to log out the current user from all the applications the user is currently logged into, bypassing the user agent.
```

You use the `quarkus.oidc.token.lifespan-grace` property only to consider some small clock skews.
====

> When the current authenticated user returns to the protected Quarkus endpoint and the ID token associated with the session cookie has expired, the user is automatically redirected to the OIDC Authorization endpoint to re-authenticate, by default.

Reviewer: This version looks like a more complex version of the previous one. I'm not sure we need to spend time on rewriting what looks like a proper sentence already. As far as I recall, the previous version was also authored by the Doc team
